.github/workflows/release.yml

```diff
@@ -19,79 +30,94 @@ jobs:
           JRELEASER_GPG_SECRET_KEY: ${{ secrets.JRELEASER_GPG_SECRET_KEY }}
-          JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME: ${{ secrets.JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME }}
-          JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD: ${{ secrets.JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD }}
+          JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME: ${{ secrets.JRELEASER_NEXUS2_USERNAME }}
+          JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD: ${{ secrets.JRELEASER_NEXUS2_PASSWORD }}
 
-      - name: JReleaser release output timefold-solver
-        if: always()
+      - name: JReleaser release output
         uses: actions/upload-artifact@v3
+        if: always()
         with:
           name: jreleaser-release
           path: |
-            timefold-solver/target/jreleaser/trace.log
-            timefold-solver/target/jreleaser/output.properties
+            target/jreleaser/trace.log
+            target/jreleaser/output.properties
 
-      - name: Set snapshot version timefold-solver
-        working-directory: ./timefold-solver
-        run: mvn --no-transfer-progress --batch-mode -Dfull versions:set -DnewVersion=${{ github.event.inputs.nextVersion }}-SNAPSHOT
+      - name: Set micro snapshot version on the release branch
+        run: |
+          mvn --no-transfer-progress --batch-mode -Dfull versions:set -DnewVersion=${{ github.event.inputs.nextMicroVersion }}-SNAPSHOT
+          find . -name 'pom.xml' | xargs git add
+          git commit -m "chore: move to ${{ github.event.inputs.nextMicroVersion }}-SNAPSHOT"
 
-      - name: Commit and push tag snapshot version timefold-solver
-        uses: actions-js/push@v1.4
-        with:
-          directory: ./timefold-solver
-          github_token: ${{ secrets.JRELEASER_GITHUB_TOKEN }}
-          message: Set next SNAPSHOT version ${{ github.event.inputs.nextVersion }}-SNAPSHOT
-          tags: true
+      - name: Switch back to source branch and set snapshot version
+        run: |
+          git checkout ${{ github.event.inputs.sourceBranch }}
+          git checkout -B ${{ github.event.inputs.releaseBranch }}-bump
+          mvn --no-transfer-progress --batch-mode -Dfull versions:set -DnewVersion=${{ github.event.inputs.nextVersion }}-SNAPSHOT
+          mvn --no-transfer-progress --batch-mode -Dfull verify -DskipTests=true 
+
+      - name: Update source branch and commit
+        run: |
+          cp docs/target/antora-template.yml docs/src/antora.yml
+          git add docs/src/antora.yml
+          find . -name 'pom.xml' | xargs git add
+          git commit -m "chore: move to ${{ github.event.inputs.nextVersion }}-SNAPSHOT"
+
+      # JReleaser should push the tag on its own, but it does not due to a bug.
+      # https://github.com/jreleaser/jreleaser/issues/1425
+      - name: Push branches to git
+        run: |
+          git push origin ${{ github.event.inputs.releaseBranch }}
+          git push origin ${{ github.event.inputs.releaseBranch }}-bump
```

Comment:
In the last stage of the release, this branch should be deleted. Apart from just having a stale branch in the git repository, if you want to do a micro release from already existing release branch, there will likely be a conflict with the existing release-bump branch.

Reply:
This needs to exist for the PR. The PR is responsible for deleting that branch; in fact, it happens automatically. I'll add it to the manual PR steps.